forage-android/src/main/java/com/joinforage/forage/android/LDManager.kt

```diff
@@ -72,17 +85,54 @@ internal object LDManager {
         }
         val contextKind = ContextKind.of(LDContextKind.SERVICE)
         val context = LDContext.create(contextKind, LDContexts.ANDROID_CONTEXT)
-        val client = LDClient.init(app, ldConfig, context, 0)
+        client = LDClient.init(app, ldConfig, context, 0)
+
+        internalLogger = logger
+    }
+
+    // vaultType is instantiated lazily and is a singleton. Once we set the vault type once, we don't
+    // want to overwrite it! We must take in the application as a parameter, which means that a
+    // ForagePINEditText must be rendered before any of the ForageSDKApi functions are called.
+    internal fun getVaultProvider(): VaultType {
+        if (vaultType != null) {
+            return vaultType as VaultType
+        }
         // default to 100% VGS usage in case LD flag retrieval fails
-        val vaultPercent = client.doubleVariation(LDFlags.VAULT_PRIMARY_TRAFFIC_PERCENTAGE_FLAG, 0.0)
+        val vaultPercent =
+            client?.doubleVariation(LDFlags.VAULT_PRIMARY_TRAFFIC_PERCENTAGE_FLAG, 0.0)
         internalLogger.i("[LaunchDarkly] Vault percent of $vaultPercent return from LD")
         val randomNum = Math.random() * 100
-        vaultType = if (randomNum < vaultPercent) {
+
+        vaultType = if (randomNum < (vaultPercent ?: 0.0)) {
             VaultType.BT_VAULT_TYPE
         } else {
             VaultType.VGS_VAULT_TYPE
         }
         internalLogger.i("[LaunchDarkly] Vault type set to $vaultType")
         return vaultType as VaultType
     }
+
+    internal fun getPollingIntervalsFeatureFlag(): IntArray {
+        if (pollingIntervals != null) {
+            return pollingIntervals as IntArray
+        }
+
+        var pollingList = IntArray(0)
+
+        val jsonIntervals =
```

Comment:
Could you do the same thing for this setter by including the default value here? Also, can you set the default value as a variable above for readability?

If the LDClient is null, I think this should work right?

```kotlin
val defaultVal = LDValue.buildObject().put("intervals", LDValue.buildArray().add(1000).build()).build()
val jsonIntervals = client?.jsonValueVariation(LDFlags.ISO_POLLING_WAIT_INTERVALS, defaultVal) ?: defaultVal
```

Reply:
yeah that works, ill make the change